lib/resourcemerge/machineconfig.go

```diff
@@ -44,6 +53,16 @@ func EnsureMachineConfigPool(modified *bool, existing *mcfgv1.MachineConfigPool,
 	}
 }
 
+func ensureMachineConfigNodeSpec(modified *bool, existing *mcfgalphav1.MachineConfigNodeSpec, required mcfgalphav1.MachineConfigNodeSpec) {
```

Comment:
comment (non-blocking): I know you were following the prevailing convention when you wrote this. Nevertheless, it is a strange convention to not return a boolean value, instead opting to mutate one instead.

(To be clear, I'm not asking you to change this given that you were following the prevailing convention.)